src/spaces/productspace.jl

```diff
@@ -151,7 +151,7 @@ function blocksectors(P::ProductSpace{S, N}) where {S, N}
     end
     bs = Vector{I}()
     if N == 0
-        push!(bs, unit(I))
+        return allunits(I)
```

Comment:
I have two remarks/questions for this:
- It is slightly odd to have a different return type for the `N == 0` case, which could be anything, instead of the sorted vector that is otherwise returned. In principle this should be fine, and the generic code paths should handle everything just fine, but it might be nicer to just be consistent and do something like the following and ensure everything is sorted
```julia
for u in allunits(I)
    push!(bs, u)
end
```
- If not, then this statement should probably be moved above the allocation of the `bs` vector